File: Java/app/src/main/java/software/rsquared/template/di/AppModule.java
package software.rsquared.template.di;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import software.rsquared.template.utils.AppExecutors;

/**
 * @author rSquared.software
 */

@Module(includes = {ViewModelModule.class})
class AppModule {

    @Provides
    @Singleton
    Context provideAppContext(Application application) {
        return application;
    }

    @Provides
    @Singleton
    AppExecutors provideAppExecutors() {
        return new AppExecutors();
    }

}

File: app/src/main/java/software/rsquared/template/di/AppModule.kt
package software.rsquared.template.di

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author Rafal Orlik
 */
@Module(includes = [ViewModelModule::class])
internal class AppModule {

    @Provides
    @Singleton
    fun provideAppContext(application: Application): Context {
        return application
    }

}